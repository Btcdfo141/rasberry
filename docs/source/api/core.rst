.. _core_module:

:mod:`spencerassistant.core`
-------------------------

.. automodule:: spencerassistant.core
    :members: