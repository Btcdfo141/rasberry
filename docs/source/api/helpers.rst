:mod:`spencerassistant.helpers`
============================

.. automodule:: spencerassistant.helpers
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.aiohttp\_client
-------------------------------------

.. automodule:: spencerassistant.helpers.aiohttp_client
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.area\_registry
------------------------------------

.. automodule:: spencerassistant.helpers.area_registry
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.check\_config
-----------------------------------

.. automodule:: spencerassistant.helpers.check_config
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.collection
--------------------------------

.. automodule:: spencerassistant.helpers.collection
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.condition
-------------------------------

.. automodule:: spencerassistant.helpers.condition
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.config\_entry\_flow
-----------------------------------------

.. automodule:: spencerassistant.helpers.config_entry_flow
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.config\_entry\_oauth2\_flow
-------------------------------------------------

.. automodule:: spencerassistant.helpers.config_entry_oauth2_flow
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.config\_validation
----------------------------------------

.. automodule:: spencerassistant.helpers.config_validation
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.data\_entry\_flow
---------------------------------------

.. automodule:: spencerassistant.helpers.data_entry_flow
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.debounce
------------------------------

.. automodule:: spencerassistant.helpers.debounce
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.deprecation
---------------------------------

.. automodule:: spencerassistant.helpers.deprecation
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.device\_registry
--------------------------------------

.. automodule:: spencerassistant.helpers.device_registry
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.discovery
-------------------------------

.. automodule:: spencerassistant.helpers.discovery
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.dispatcher
--------------------------------

.. automodule:: spencerassistant.helpers.dispatcher
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.entity
----------------------------

.. automodule:: spencerassistant.helpers.entity
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.entity\_component
---------------------------------------

.. automodule:: spencerassistant.helpers.entity_component
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.entity\_platform
--------------------------------------

.. automodule:: spencerassistant.helpers.entity_platform
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.entity\_registry
--------------------------------------

.. automodule:: spencerassistant.helpers.entity_registry
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.entity\_values
------------------------------------

.. automodule:: spencerassistant.helpers.entity_values
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.entityfilter
----------------------------------

.. automodule:: spencerassistant.helpers.entityfilter
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.event
---------------------------

.. automodule:: spencerassistant.helpers.event
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.icon
--------------------------

.. automodule:: spencerassistant.helpers.icon
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.integration\_platform
-------------------------------------------

.. automodule:: spencerassistant.helpers.integration_platform
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.intent
----------------------------

.. automodule:: spencerassistant.helpers.intent
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.json
--------------------------

.. automodule:: spencerassistant.helpers.json
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.location
------------------------------

.. automodule:: spencerassistant.helpers.location
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.logging
-----------------------------

.. automodule:: spencerassistant.helpers.logging
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.network
-----------------------------

.. automodule:: spencerassistant.helpers.network
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.restore\_state
------------------------------------

.. automodule:: spencerassistant.helpers.restore_state
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.script
----------------------------

.. automodule:: spencerassistant.helpers.script
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.service
-----------------------------

.. automodule:: spencerassistant.helpers.service
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.signal
-----------------------------

.. automodule:: spencerassistant.helpers.signal
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.state
---------------------------

.. automodule:: spencerassistant.helpers.state
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.storage
-----------------------------

.. automodule:: spencerassistant.helpers.storage
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.sun
-------------------------

.. automodule:: spencerassistant.helpers.sun
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.system\_info
----------------------------------

.. automodule:: spencerassistant.helpers.system_info
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.temperature
---------------------------------

.. automodule:: spencerassistant.helpers.temperature
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.template
------------------------------

.. automodule:: spencerassistant.helpers.template
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.translation
---------------------------------

.. automodule:: spencerassistant.helpers.translation
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.typing
----------------------------

.. automodule:: spencerassistant.helpers.typing
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.update\_coordinator
-----------------------------------------

.. automodule:: spencerassistant.helpers.update_coordinator
   :members:
   :undoc-members:
   :show-inheritance:
