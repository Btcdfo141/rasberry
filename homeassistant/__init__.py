"""Init file for Home Assistant."""


test2
