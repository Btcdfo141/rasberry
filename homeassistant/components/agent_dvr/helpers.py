"""Helpers for Agent DVR component."""
from .const import DOMAIN as AGENT_DOMAIN


def service_signal(service, ident=None):
    """Encode service and identifier into signal."""
    signal = f"{AGENT_DOMAIN}_{service}"
    if ident:
        signal += "_{}".format(ident.replace(".", "_"))
    return signal


def generate_url(host, port) -> str:
    """Create a URL from the host and port."""
    server_origin = host
    if "://" not in host:
        server_origin = f"http://{host}"

    if server_origin[-1] == "/":
        server_origin = server_origin[:-1]

    return f"{server_origin}:{port}/"
