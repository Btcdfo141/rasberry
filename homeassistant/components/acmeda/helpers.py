"""Helper functions for Acmeda Pulse."""
from homeassistant.helpers.device_registry import async_get_registry as get_dev_reg
from homeassistant.helpers.entity_registry import async_get_registry as get_ent_reg

from .const import DOMAIN, LOGGER


async def update_entities(
    hass, entity_class, config_entry, current, async_add_entities
):
    """Add any new entities, remove any old ones."""
    hub = hass.data[DOMAIN][config_entry.entry_id]
    LOGGER.debug("Looking for new %s on: %s", entity_class.__name__, hub.host)

    api = hub.api.rollers

    new_items = []
    for unique_id, roller in api.items():
        if unique_id not in current:
            LOGGER.debug("New %s %s", entity_class.__name__, unique_id)
            new_item = entity_class(hass, roller)
            current[unique_id] = new_item
            new_items.append(new_item)

    async_add_entities(new_items)

    LOGGER.debug("Looking for removed %s on: %s", entity_class.__name__, hub.host)

    removed_items = []
    for unique_id, element in current.items():
        if unique_id not in api:
            LOGGER.debug("Removing %s %s", entity_class.__name__, unique_id)
            removed_items.append(element)

    for entity in removed_items:
        # Device is removed from Pulse, so we remove it from Home Assistant

        await entity.async_remove()
        ent_registry = await get_ent_reg(hass)
        if entity.entity_id in ent_registry.entities:
            ent_registry.async_remove(entity.entity_id)
        dev_registry = await get_dev_reg(hass)
        device = dev_registry.async_get_device(
            identifiers={(DOMAIN, entity.device_id)}, connections=set()
        )
        if device is not None:
            dev_registry.async_update_device(
                device.id, remove_config_entry_id=config_entry.entry_id
            )
        del current[entity.unique_id]


async def update_devices(hass, config_entry, api):
    """Tell hass that device info has been updated."""
    dev_registry = await get_dev_reg(hass)

    for api_item in api.values():
        # Update Device name
        device = dev_registry.async_get_device(
            identifiers={(DOMAIN, api_item.id)}, connections=set()
        )
        if device is not None:
            dev_registry.async_update_device(
                device.id, name=api_item.name,
            )
