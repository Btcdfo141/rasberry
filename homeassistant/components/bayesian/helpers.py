"""Helpers to deal with bayesian observations."""
from __future__ import annotations

from homeassistant.helpers.template import Template


class Observation:
    """Representation of a sensor or template observation."""

    def __init__(
        self,
        # identifier: str,
        entity_id: str | None,
        platform: str,
        prob_given_true: float,
        prob_given_false: float,
        observed: bool | None,
        to_state: str,
        above: float | None,
        below: float | None,
        value_template: Template | None,
    ) -> None:
        """Initialize the Observation."""
        # self.identifier = identifier
        self.entity_id = entity_id
        self.platform = platform
        self.prob_given_true = prob_given_true
        self.prob_given_false = prob_given_false
        self.observed = observed
        self.to_state = to_state
        self.below = below
        self.above = above
        self.value_template = value_template
        self.id: str | None = None

    def to_dict(self) -> dict[str, str | float | bool | None]:
        """Represent Class as a Dict for easier serialization."""

        return {
            "entity_id": self.entity_id,
            "platform": self.platform,
            "prob_given_true": self.prob_given_true,
            "prob_given_false": self.prob_given_false,
            "observed": self.observed,
            "to_state": self.to_state,
            "below": self.below,
            "above": self.above,
            "value_template": str(self.value_template),
            "id": self.id,
        }
