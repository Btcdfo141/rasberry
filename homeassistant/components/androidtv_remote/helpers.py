"""Helper functions for Android TV Remote integration."""
from __future__ import annotations

from androidtvremote2 import AndroidTVRemote

from homeassistant.core import HomeAssistant


def create_api(hass: HomeAssistant, host: str) -> AndroidTVRemote:
    """Create an AndroidTVRemote instance."""
    return AndroidTVRemote(
        client_name="Home Assistant",
        certfile=hass.config.path("androidtv_remote_cert.pem"),
        keyfile=hass.config.path("androidtv_remote_key.pem"),
        host=host,
        loop=hass.loop,
    )
