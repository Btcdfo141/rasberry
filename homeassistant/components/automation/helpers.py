"""
homeassistant.components.automation.helpers
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Offers helper functions for automation rules.

"""
import logging

from homeassistant.helpers import service

CONF_ENTITY_ID = "entity_id"
CONF_EXPAND_IDS = "expand_ids"

_LOGGER = logging.getLogger(__name__)


def get_entity_ids(hass, config):
    """ Get entity ids from the conf and expand if wanted. """
    entity_id = config.get(CONF_ENTITY_ID)
    expand = config.get(CONF_EXPAND_IDS, None)

    if entity_id is None:
        _LOGGER.error("Missing configuration key %s", CONF_ENTITY_ID)
        return None

    if expand:
        entity_id_old = entity_id
        entity_id = service.expand_entity_ids(hass, entity_id=entity_id)
        _LOGGER.debug("Expanded entity_ids %s to %s", entity_id_old,
                      entity_id)

    return entity_id
