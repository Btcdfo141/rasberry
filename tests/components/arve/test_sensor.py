"""Test for Arve sensors."""

from unittest.mock import MagicMock

from syrupy import SnapshotAssertion

from homeassistant.core import HomeAssistant
from homeassistant.helpers import entity_registry as er

from . import async_init_integration

from tests.common import MockConfigEntry

SENSORS = (
    "air_quality_index",
    "carbon_dioxide",
    "humidity",
    "pm10",
    "pm25",
    "temperature",
    "none",
)


async def test_sensors(
    hass: HomeAssistant,
    mock_arve: MagicMock,
    entity_registry: er.EntityRegistry,
    mock_config_entry: MockConfigEntry,
    snapshot: SnapshotAssertion,
) -> None:
    """Test the Arve sensors."""

    sn = mock_arve.serial_number.split("-")[-1]

    await async_init_integration(hass, mock_config_entry)

    for sensor in SENSORS:
        state = hass.state.get(f"sensor.sensor_{sn}_{sensor}")
        assert state
        assert state == snapshot(name=f"sensor_{sn}_{sensor}")

        entry = entity_registry.async_get(state.entity_id)
        assert entry
        assert entry.device_id
        assert entry == snapshot(name=f"entry_{sn}_{sensor}")
