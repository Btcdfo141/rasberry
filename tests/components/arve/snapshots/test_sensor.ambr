# serializer version: 1
# name: test_sensors[entry_test-serial-number_air_quality_index]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'sensor',
    'entity_category': None,
    'entity_id': 'sensor.my_arve_air_quality_index',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': <SensorDeviceClass.AQI: 'aqi'>,
    'original_icon': None,
    'original_name': 'AQI',
    'platform': 'arve',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': 'aqi',
    'unique_id': 'test-serial-number_aqi',
    'unit_of_measurement': None,
  })
# ---
# name: test_sensors[entry_test-serial-number_carbon_dioxide]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'sensor',
    'entity_category': None,
    'entity_id': 'sensor.my_arve_carbon_dioxide',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': <SensorDeviceClass.CO2: 'carbon_dioxide'>,
    'original_icon': None,
    'original_name': 'CO2',
    'platform': 'arve',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': 'co2',
    'unique_id': 'test-serial-number_co2',
    'unit_of_measurement': 'ppm',
  })
# ---
# name: test_sensors[entry_test-serial-number_humidity]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'sensor',
    'entity_category': None,
    'entity_id': 'sensor.my_arve_humidity',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': <SensorDeviceClass.HUMIDITY: 'humidity'>,
    'original_icon': None,
    'original_name': 'Humidity',
    'platform': 'arve',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': 'humidity',
    'unique_id': 'test-serial-number_humidity',
    'unit_of_measurement': '%',
  })
# ---
# name: test_sensors[entry_test-serial-number_none]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'sensor',
    'entity_category': None,
    'entity_id': 'sensor.my_arve_none',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': None,
    'original_name': 'TVOC',
    'platform': 'arve',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': 'tvoc',
    'unique_id': 'test-serial-number_tvoc',
    'unit_of_measurement': None,
  })
# ---
# name: test_sensors[entry_test-serial-number_pm10]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'sensor',
    'entity_category': None,
    'entity_id': 'sensor.my_arve_pm10',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': <SensorDeviceClass.PM10: 'pm10'>,
    'original_icon': None,
    'original_name': 'PM10',
    'platform': 'arve',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': 'pm10',
    'unique_id': 'test-serial-number_pm10',
    'unit_of_measurement': 'µg/m³',
  })
# ---
# name: test_sensors[entry_test-serial-number_pm2_5]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'sensor',
    'entity_category': None,
    'entity_id': 'sensor.my_arve_pm2_5',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': <SensorDeviceClass.PM25: 'pm25'>,
    'original_icon': None,
    'original_name': 'PM25',
    'platform': 'arve',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': 'pm25',
    'unique_id': 'test-serial-number_pm25',
    'unit_of_measurement': 'µg/m³',
  })
# ---
# name: test_sensors[entry_test-serial-number_temperature]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'sensor',
    'entity_category': None,
    'entity_id': 'sensor.my_arve_temperature',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': <SensorDeviceClass.TEMPERATURE: 'temperature'>,
    'original_icon': None,
    'original_name': 'Temperature',
    'platform': 'arve',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': 'temperature',
    'unique_id': 'test-serial-number_temperature',
    'unit_of_measurement': <UnitOfTemperature.CELSIUS: '°C'>,
  })
# ---
# name: test_sensors[entry_test-serial-number_tvoc]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'sensor',
    'entity_category': None,
    'entity_id': 'sensor.my_arve_tvoc',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': None,
    'original_name': 'TVOC',
    'platform': 'arve',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': 'tvoc',
    'unique_id': 'test-serial-number_tvoc',
    'unit_of_measurement': None,
  })
# ---
# name: test_sensors[my_arve_air_quality_index]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'device_class': 'aqi',
      'friendly_name': 'My Arve AQI',
    }),
    'context': <ANY>,
    'entity_id': 'sensor.my_arve_air_quality_index',
    'last_changed': <ANY>,
    'last_updated': <ANY>,
    'state': "<AsyncMock name='Arve().device_sensor_data().aqi' id='4673889600'>",
  })
# ---
# name: test_sensors[my_arve_carbon_dioxide]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'device_class': 'carbon_dioxide',
      'friendly_name': 'My Arve CO2',
      'unit_of_measurement': 'ppm',
    }),
    'context': <ANY>,
    'entity_id': 'sensor.my_arve_carbon_dioxide',
    'last_changed': <ANY>,
    'last_updated': <ANY>,
    'state': "<AsyncMock name='Arve().device_sensor_data().co2' id='4683517632'>",
  })
# ---
# name: test_sensors[my_arve_humidity]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'device_class': 'humidity',
      'friendly_name': 'My Arve Humidity',
      'unit_of_measurement': '%',
    }),
    'context': <ANY>,
    'entity_id': 'sensor.my_arve_humidity',
    'last_changed': <ANY>,
    'last_updated': <ANY>,
    'state': "<AsyncMock name='Arve().device_sensor_data().humidity' id='4674090384'>",
  })
# ---
# name: test_sensors[my_arve_none]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'My Arve TVOC',
    }),
    'context': <ANY>,
    'entity_id': 'sensor.my_arve_none',
    'last_changed': <ANY>,
    'last_updated': <ANY>,
    'state': "<AsyncMock name='Arve().device_sensor_data().tvoc' id='4430967152'>",
  })
# ---
# name: test_sensors[my_arve_pm10]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'device_class': 'pm10',
      'friendly_name': 'My Arve PM10',
      'unit_of_measurement': 'µg/m³',
    }),
    'context': <ANY>,
    'entity_id': 'sensor.my_arve_pm10',
    'last_changed': <ANY>,
    'last_updated': <ANY>,
    'state': "<AsyncMock name='Arve().device_sensor_data().pm10' id='4683800288'>",
  })
# ---
# name: test_sensors[my_arve_pm2_5]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'device_class': 'pm25',
      'friendly_name': 'My Arve PM25',
      'unit_of_measurement': 'µg/m³',
    }),
    'context': <ANY>,
    'entity_id': 'sensor.my_arve_pm2_5',
    'last_changed': <ANY>,
    'last_updated': <ANY>,
    'state': "<AsyncMock name='Arve().device_sensor_data().pm25' id='4683861792'>",
  })
# ---
# name: test_sensors[my_arve_temperature]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'device_class': 'temperature',
      'friendly_name': 'My Arve Temperature',
      'unit_of_measurement': <UnitOfTemperature.CELSIUS: '°C'>,
    }),
    'context': <ANY>,
    'entity_id': 'sensor.my_arve_temperature',
    'last_changed': <ANY>,
    'last_updated': <ANY>,
    'state': "<AsyncMock name='Arve().device_sensor_data().temperature' id='4683873744'>",
  })
# ---
# name: test_sensors[my_arve_tvoc]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'My Arve TVOC',
    }),
    'context': <ANY>,
    'entity_id': 'sensor.my_arve_tvoc',
    'last_changed': <ANY>,
    'last_updated': <ANY>,
    'state': "<AsyncMock name='Arve().device_sensor_data().tvoc' id='4683902432'>",
  })
# ---
